src/components/buttons.tsx

```diff
@@ -1,54 +1,93 @@
 import { colors } from '@hedviginsurance/brand'
 import styled from 'react-emotion'
+import { colorComponentColors } from 'src/blocks/BaseBlockProps'
 
 const buttonSizes = {
   sm: '.75rem 1.5rem',
   md: '1rem 2rem',
 }
 
 export type ButtonStyleType = 'filled' | 'outlined' | 'plain'
+export type ButtonWeight = 'normal' | 'bold'
 
 interface ButtonProps {
   size?: keyof typeof buttonSizes
-  bold?: boolean
   styleType?: ButtonStyleType
+  color?: colorComponentColors
+  weight?: ButtonWeight
 }
 
-const buttonTypeStyles = {
-  filled: {
-    backgroundColor: colors.GREEN,
-    color: colors.WHITE,
-  },
-  outlined: {
-    backgroundColor: 'transparant',
-    color: colors.GREEN,
-  },
-  plain: {
-    border: 'none',
-    padding: '0',
-    backgroundColor: 'transparant',
-    color: colors.GREEN,
-  },
+type ColorMap = { [K in colorComponentColors]: string }
```

Comment:
i guess also this could be something like `Record<colorComponentColors, string>` if we want :)